### PR TITLE
Use a selector to mount and unmount components at a given node

### PR DESCRIPTION
**Pull request by jakegavin**
I'm working on an app where we need to be able to manually unmount specific components when a modal is closed. This PR adds a `searchSelector` argument to `ReactRailsUJS.mountComponents` and `ReactRailsUJS.unmountComponents` which lets you target specific elements. 

This is similar to https://github.com/reactjs/react-rails/pull/91. This functionality has been listed as a goal for 1.0 (https://github.com/reactjs/react-rails/issues/133) but the author of #91 hasn't responded in two months so I opened this PR.

### CODE REVIEW

**Comment by danott** [comment]
:+1: I've found myself reaching for this exact behavior multiple times.

**Comment by rmosolgo** [comment]
Nice feature! Looks good to me, let me just give it a try locally with Turbolinks and without

**Comment by rmosolgo** [comment]
Looks good in a local test app w/Turbolinks & without, and looks good in PCO Check-ins

**Comment by rmosolgo** [comment]
:tada: :confetti_ball:  thanks!

File: test/view_helper_test.rb [hunk 1]
```diff
@@ -115,13 +115,24 @@ class ViewHelperTest < ActionDispatch::IntegrationTest
     assert page.has_content?('Hello Bob')
   end
 
+  test 'react_ujs can unount at node' do
```
Comment (vipulnsward) [inline test/view_helper_test.rb]: `unmount`

**Pull request merged**